common.py

```diff
@@ -45,23 +55,72 @@ def nested_process_search(self, criteria, cb_conn, base_query):
 
         elif self.product == "cbth":
             return cbth.nested_process_search(cb_conn, criteria, base_query)
+        
+        elif self.product == "defender": 
+            return defender.nested_process_search(cb_conn, criteria, base_query)
 
     # write the rows of the CSV
     def write_csv(self, output, results, *args):
         for r in results:
             row = [r[0], r[1], r[2], r[3], args[0], args[1]]
             output.writerow(row)
 
-    def get_cbapi_connection(self):
+    def get_connection(self):
         if self.product == 'cbr':
             if self.profile:
                 cb_conn = CbEnterpriseResponseAPI(profile=self.profile)
             else:
                 cb_conn = CbEnterpriseResponseAPI()
 
+            return cb_conn
+
         elif self.product == 'cbth':
             if self.profile:
                 cb_conn = CbThreatHunterAPI(profile=self.profile)
             else:
                 cb_conn = CbThreatHunterAPI()
-        return cb_conn
\ No newline at end of file
+            
+            return cb_conn
+
+    def get_connection_creds(self, creds):
```

Comment:
🗣️ Is the intent with this method to provide a source to fetch other connections beyond defender given a set of credentials?  That seems good and worth doing but maybe the `atp_profile` setting should get moved under the `if self.product == 'defender':` block so it's grouped together.

What's the expected behavior if the product is not defender?  Raise something or warn?

Reply:
Right now this method will only get called if the ```product == defender``` (in surveyor.py line 73-74). The idea was that in the future if there are other platforms that don't have standard locations for credentials like ```cbapi``` we can just add them here. If it's a CB product it should call the ```get_connection()```. 

I added the profile pieces to under the ```if self.product == 'defender':``` block.